tuner/tuner/libtuner.py

```diff
@@ -908,6 +1121,87 @@ def collision_handler(index_hash_list: list[tuple[int, str]]) -> tuple[bool, lis
     return collision_detected, unique_indexes
 
 
+def compile(
+    args: argparse.Namespace,
+    path_config: PathConfig,
+    candidates: list[int],
+    candidate_trackers: list[CandidateTracker],
+    tuning_client: TuningClient,
+    input_file: Optional[Path] = None,
+) -> list[int]:
+    logging.debug("compile()")
+
+    if not candidates:
+        logging.warning("No model candidates to compile.")
+        return []
+
+    # If `input_file` is not None, then replace the currently tracked template
+    # with the passed input mlir file.
+    if input_file is not None:
+        shutil.copy(input_file, path_config.template_mlir)
+
+    # Strip compilation info and root_op attribute from the source and save
+    # the stripped IR, since the TD specs do not expect these attributes.
+    stripped_mlir = strip_compilation_info(path_config.template_mlir)
+    with ir.Context():
```

Comment:
Why are we creating a new context instead of using the one from the tuning client?

Reply:
The context isn't actually being used for anything other than to make the parser happy here. This is going from input_file->output_file, so the context is completely local to this one section of code. Also, afaict, there is no guarantee that the tuning client has a context?

Reply:
I see, then I can add a context to the TuningClient class and thread it through.

Reply:
Feel free to add a TODO, it's not super important